### DISCUSSION TITLE
Websocket in Routers

### DISCUSSION THREAD

**Discussion by AliAryaiee** [fastapi/fastapi#8594] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# main.py
from random import randint
from fastapi import FastAPI, WebSocket

app = FastAPI()

@app.get("/", tags=["Test"])
async def index():
    return {
        "response": "Index"
    }


@app.websocket("/wsx")
async def test_socket(websocket: WebSocket, start: int = 1, stop: int = 100):
    await websocket.accept()
    while True:
        try:
            number = await websocket.receive_text()
            rndm = randint(start, stop)
            response = int(number) + rndm
            await websocket.send_json(
                {"number": response}
            )
        except Exception as error:
            break


# router.py
from random import randint
from fastapi import APIRouter, WebSocket

socket_router = APIRouter(
    prefix="/wsx",
    tags=["Socket"]
)


@socket_router.websocket("/")
async def test_socket(websocket: WebSocket, start: int = 1, stop: int = 100):
    await websocket.accept()
    while True:
        try:
            number = await websocket.receive_text()
            rndm = randint(start, stop)
            response = int(number) + rndm
            await websocket.send_json(
                {"number": response}
            )
        except Exception as error:
            break
```

### Description

When I use this WebSocket inside of "mani.py", everything is Ok.
But when I move this WebSocket from "main.py" to "router.py" and I use APIRouter in "router.py" and then I include this router inside of "main.py", WebSocket doesn't work!

what's the solution

### Operating System

Windows

### FastAPI Version

0.73.0

### Python Version

3.7.9

**Comment by Kludex** [comment]
Duplicated

**Comment by srcortez45** [comment]
i had the same problem, then i created class manager websocket

**Comment by Kludex** [comment, marked as answer]
This issue was already solved.
